### PR TITLE
add Date and Message-ID to mail

### PR DESCRIPTION
**Pull request by my3**
**Issue:**   #569

### CODE REVIEW

File: app/pkg/email/smtp/smtp.go [hunk 1]
```diff
@@ -157,3 +168,15 @@ func sendMail(localName, serverAddress string, a gosmtp.Auth, from string, to []
 	}
 	return c.Quit()
 }
+
+func generateMessageID() (string, error) {
+	timestamp := strconv.FormatInt(time.Now().UTC().UnixNano(), 10)
+	unq := uuid.New()
+	randStr := unq.String()
+	host, err := os.Hostname()
```
Comment (goenning) [inline app/pkg/email/smtp/smtp.go]: We shouldn't use os.Hostname here, the hostname is usually completely different than the real domain name. There's a `localname` variable that we could use instead.

File: app/pkg/email/smtp/smtp.go [hunk 2]
```diff
@@ -157,3 +168,15 @@ func sendMail(localName, serverAddress string, a gosmtp.Auth, from string, to []
 	}
 	return c.Quit()
 }
+
+func generateMessageID() (string, error) {
+	timestamp := strconv.FormatInt(time.Now().UTC().UnixNano(), 10)
+	unq := uuid.New()
```
Comment (goenning) [inline app/pkg/email/smtp/smtp.go]: I'd prefer to use `rand.String(32)` that is already available on Fider. This random string + timestamp is unique enough for this operation. By doing so, we won't need the external `uuid` package.

File: app/pkg/email/smtp/smtp.go [hunk 3]
```diff
@@ -157,3 +163,12 @@ func sendMail(localName, serverAddress string, a gosmtp.Auth, from string, to []
 	}
 	return c.Quit()
 }
+
+func generateMessageID(localName string) string {
+	timestamp := strconv.FormatInt(time.Now().UTC().UnixNano(), 10)
+	buf := make([]byte, 16)
+	rand.Read(buf)
```
Comment (goenning) [inline app/pkg/email/smtp/smtp.go]: `rand.Read` might throw an error, that's very unlikely to happen, but if does, we should panic. We have a top `recover` goroutine that handles panics.

```suggestion
	_, err := rand.Read(buf)
	if err != nil {
	  panic(err)
	}
```

File: Gopkg.lock [hunk 4]
```diff
@@ -173,6 +173,7 @@
     "github.com/dgrijalva/jwt-go",
     "github.com/goenning/letteravatar",
     "github.com/goenning/sqlcertcache",
+    "github.com/google/uuid",
```
Comment (goenning) [inline Gopkg.lock]: Just revert all changes to Gopkg.lock

File: Gopkg.toml [hunk 5]
```diff
@@ -37,3 +37,4 @@ ignored = ["github.com/magefile/mage/mg","github.com/magefile/mage/sh"]
   branch = "master"
   name = "github.com/goenning/sqlcertcache"
 
+
```
Comment (goenning) [inline Gopkg.toml]: Just revert all changes to Gopkg.toml

File: Gopkg.lock (hunk 4 above)
Comment (my3) [reply Gopkg.lock]: ah.. thought I reverted all changes. will update and squash commits or create a new PR with one commit

**Comment by goenning** [comment]
Thank you so much for this! 🎉

**Pull request merged**